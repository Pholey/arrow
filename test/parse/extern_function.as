extern def method();

extern def method(): int64;

extern def method(a: int64, b: int64): bool;
