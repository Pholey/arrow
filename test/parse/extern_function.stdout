{
  "tag": "Module",
  "span": "test/parse/extern_function.as:1-5",
  "statements": [
    {
      "tag": "ExternFunction",
      "span": "test/parse/extern_function.as:1:1-20",
      "name": "method",
      "result_type": null,
      "parameters": []
    },
    {
      "tag": "ExternFunction",
      "span": "test/parse/extern_function.as:3:1-27",
      "name": "method",
      "result_type": {
        "tag": "TypeName",
        "span": "test/parse/extern_function.as:3:22-27",
        "name": {
          "tag": "Name",
          "span": "test/parse/extern_function.as:3:22-27",
          "text": "int64",
          "type_arguments": []
        }
      },
      "parameters": []
    },
    {
      "tag": "ExternFunction",
      "span": "test/parse/extern_function.as:5:1-44",
      "name": "method",
      "result_type": {
        "tag": "TypeName",
        "span": "test/parse/extern_function.as:5:40-44",
        "name": {
          "tag": "Name",
          "span": "test/parse/extern_function.as:5:40-44",
          "text": "bool",
          "type_arguments": []
        }
      },
      "parameters": [
        {
          "tag": "Parameter",
          "span": "test/parse/extern_function.as:5:19-27",
          "name": "a",
          "type": {
            "tag": "TypeName",
            "span": "test/parse/extern_function.as:5:22-27",
            "name": {
              "tag": "Name",
              "span": "test/parse/extern_function.as:5:22-27",
              "text": "int64",
              "type_arguments": []
            }
          }
        },
        {
          "tag": "Parameter",
          "span": "test/parse/extern_function.as:5:29-37",
          "name": "b",
          "type": {
            "tag": "TypeName",
            "span": "test/parse/extern_function.as:5:32-37",
            "name": {
              "tag": "Name",
              "span": "test/parse/extern_function.as:5:32-37",
              "text": "int64",
              "type_arguments": []
            }
          }
        }
      ]
    }
  ]
}
